cvmfs/catalog_mgr_rw.h

```diff
@@ -160,8 +174,72 @@ class WritableCatalogManager : public SimpleCatalogManager {
    * Directories don't have extended attributes at this point.
    */
   XattrList empty_xattrs;
+
+  /**
+   * Balancer thresholds
+   */
+  const bool is_balanced_;
+  const unsigned max_weight_;
+  const unsigned min_weight_;
+  const unsigned balance_weight_;
 };  // class WritableCatalogManager
 
+
+template <class CatalogMgrT>
+class CatalogBalancer {
+ public:
+  typedef typename CatalogMgrT::catalog_t catalog_t;
+  explicit CatalogBalancer(CatalogMgrT *catalog_mgr)
+    : catalog_mgr_(catalog_mgr) { }
+
+  void Balance(catalog_t *catalog = NULL);
+
+
+ protected:
```

Comment:
Yes, I put it protected in case one day somebody would like to create different CatalogBalancers using different techniques, but right now it could perfectly be private.